packages/framework/src/Component/ClassExtension/MethodAnnotationsFactory.php

```diff
@@ -63,16 +62,15 @@ public function getMethodAnnotationLine(
         ReflectionClass $projectClassBetterReflection
     ): string {
         foreach ($this->annotationsReplacementsMap->getPatterns() as $frameworkClassPattern) {
-            foreach ($reflectionMethodFromFrameworkClass->getDocBlockReturnTypes() as $docBlockReturnType) {
-                if (!$this->isMethodImplementedInClass($reflectionMethodFromFrameworkClass->getName(), $projectClassBetterReflection)
-                    && $this->isMethodReturningTypeThatIsExtendedInProject($frameworkClassPattern, $docBlockReturnType)
-                ) {
+            if (!$this->isMethodImplementedInClass($reflectionMethodFromFrameworkClass->getName(), $projectClassBetterReflection)) {
+                if ($this->isMethodReturningTypeThatIsExtendedInProject($frameworkClassPattern, $reflectionMethodFromFrameworkClass->getDocBlockReturnTypes())
+                    || $this->methodParameterTypeIsExtendedInProject($frameworkClassPattern, $reflectionMethodFromFrameworkClass->getParameters())) {
                     return sprintf(
-                        " * @method %s%s %s(%s)\n",
+                        " * @method %s%s%s(%s)\n",
                         $reflectionMethodFromFrameworkClass->isStatic() ? 'static ' : '',
-                        $this->annotationsReplacer->replaceInMethodReturnType($reflectionMethodFromFrameworkClass),
+                        !empty($this->annotationsReplacer->replaceInMethodReturnType($reflectionMethodFromFrameworkClass)) ? $this->annotationsReplacer->replaceInMethodReturnType($reflectionMethodFromFrameworkClass) . ' ' : '',
                         $reflectionMethodFromFrameworkClass->getName(),
-                        $this->getMethodParameterNames($reflectionMethodFromFrameworkClass)
+                        $this->getMethodParameterNamesWithTypes($reflectionMethodFromFrameworkClass)
```

Comment:
I'm starting to be a little lost with these lines.
What about
```php
$optionalStaticKeyword = $reflectionMethodFromFrameworkClass->isStatic() ? 'static ' : '';
$returnType = !empty($this->annotationsReplacer->replaceInMethodReturnType($reflectionMethodFromFrameworkClass)) ? $this->annotationsReplacer->replaceInMethodReturnType($reflectionMethodFromFrameworkClass) . ' ' : '';
$methodName = $reflectionMethodFromFrameworkClass->getName();
$parameterNamesWithTypes = $this->getMethodParameterNamesWithTypes($reflectionMethodFromFrameworkClass);

// maybe better variable names

return sprintf(" * @method %s%s%s(%s)\n",
    $optionalStaticKeyword,
    $returnType,
    $methodName,
    $parameterNamesWithTypes
);
```

So it's clear what is created here. What do you think?

Reply:
love it :ok_hand: applied in c58460133